test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -2083,12 +2083,17 @@ def decode_value(cls, value):
         of it, if it is made of alphanumeric values only, return
         it as a string, if it is of type size, i.e., with a memory
         unit such as b,kb,mb,gb then return the converted size to
-        kb without the unit
+        kb without the unit, if value is a dictionary string return
+        dictionary value
 
         :param value: attribute/resource value
         :type value: str or int
         :returns: int or float or string
         """
+        if re.search('(\{.*:.*[,]?.*\})', str(value)):
+            value = ast.literal_eval(ast.literal_eval(str(value)))
```

Comment:
Why do you need to call it twice? Calling it once works for me:

```
>>> value='{1:"blah", 2:[1, "a", "blah"], 3: {1:2, 3:4}}'
>>> ast.literal_eval(value)
{1: 'blah', 2: [1, 'a', 'blah'], 3: {1: 2, 3: 4}}
```

^ The value above is already a dict type now, not string. Calling it twice infact errors out:

```
>>> ast.literal_eval(ast.literal_eval(value))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/ast.py", line 91, in literal_eval
    return _convert(node_or_string)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/ast.py", line 90, in _convert
    return _convert_signed_num(node)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/ast.py", line 63, in _convert_signed_num
    return _convert_num(node)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.7/lib/python3.7/ast.py", line 55, in _convert_num
    raise ValueError('malformed node or string: ' + repr(node))
ValueError: malformed node or string: {1: 'blah', 2: [1, 'a', 'blah'], 3: {1: 2, 3: 4}}
```

Reply:
I have added test results, those infact work fine.
The value passed in the function is actually this:
`value='\'{1:"blah", 2:[1, "a", "blah"], 3: {1:2, 3:4}}\''`

So first time the method is called it evaluates to string, and second time to dict

Reply:
That's not really a dict in string form though right? That's just a string. The function doesn't decode \'\2\'' to numeric 2, so I don't think it should decode '\'{1:"blah", 2:[1, "a", "blah"], 3: {1:2, 3:4}}\'' to a dict. Where is this value coming from? If the test creates it then it should remove the escaped quotes before passing it to this function.

Reply:
I agree with @agrawalravi90 here.